### PR TITLE
add warning for theano_config

### PR DESCRIPTION
**Pull request by ferrine**
Adds warning following #2378
CC @aseyboldt

### CODE REVIEW

File: pymc3/model.py [hunk 1]
```diff
@@ -366,9 +367,17 @@ def __new__(cls, *args, **kwargs):
...
+        theano_config = kwargs.get('theano_config', {})
+        if 'compute_test_value' not in theano_config:
+            theano_config['compute_test_value'] = 'raise'
+        if theano_config['compute_test_value'] != theano.config.compute_test_value:
+            warnings.warn("You are using theano config "
+                          "'compute_test_value':%r while Model requires "
+                          "'compute_test_value':%r. The latter is forced "
+                          "in context manager, we suggest you to use the "
+                          "same values to avoid weird bugs"
+                          % (theano.config.compute_test_value, theano_config['compute_test_value']),
```
Comment (twiecki) [inline pymc3/model.py]: better to use new-style `"".format()`.

File: pymc3/model.py [hunk 2]
```diff
@@ -366,9 +367,17 @@ def __new__(cls, *args, **kwargs):
...
-            theano_config = {'compute_test_value': 'raise'}
+        theano_config = kwargs.get('theano_config', {})
+        if 'compute_test_value' not in theano_config:
+            theano_config['compute_test_value'] = 'raise'
+        if theano_config['compute_test_value'] != theano.config.compute_test_value:
+            warnings.warn("You are using theano config "
+                          "'compute_test_value':%r while Model requires "
+                          "'compute_test_value':%r. The latter is forced "
+                          "in context manager, we suggest you to use the "
+                          "same values to avoid weird bugs"
```
Comment (twiecki) [inline pymc3/model.py]: we suggest you to set compute_test_value to 'raise' to avoid weird behavior

File: pymc3/model.py [hunk 3]
```diff
@@ -366,9 +367,17 @@ def __new__(cls, *args, **kwargs):
             instance._parent = cls.get_contexts()[-1]
         else:
             instance._parent = None
-        theano_config = kwargs.get('theano_config', None)
-        if theano_config is None or 'compute_test_value' not in theano_config:
-            theano_config = {'compute_test_value': 'raise'}
+        theano_config = kwargs.get('theano_config', {})
+        if 'compute_test_value' not in theano_config:
+            theano_config['compute_test_value'] = 'raise'
+        if theano_config['compute_test_value'] != theano.config.compute_test_value:
```
Comment (aseyboldt) [inline pymc3/model.py]: Shouldn't this check be `elif theano_config['compute_test_value'] not in ['raise', 'ignore']`? I don't see why we would care about the current state in theano. And we can't run this if that key is not in `theano_config`.

File: pymc3/model.py (hunk 1 above)
Comment (aseyboldt) [reply pymc3/model.py]: `(theano_config['compute_test_value']), 'raise')`, right?

File: pymc3/model.py (hunk 3 above)
Comment (ferrine) [reply pymc3/model.py]: It is set above if not provided

Comment (aseyboldt) [reply pymc3/model.py]: Sorry, I see. But I still don't see why you would check it agains the current value.

Comment (ferrine) [reply pymc3/model.py]: In previous version it had the same behavior. User could provide any flag

Comment (ferrine) [reply pymc3/model.py]: Different flags matter. Equal are expected behavior

Comment (ferrine) [reply pymc3/model.py]: But you are right that we need 'raise' only

Comment (aseyboldt) [reply pymc3/model.py]: But we should warn users if they set the value to something that doesn't compute a testval at all, and not when they change the flag. Usually, `theano.config.compute_test_value` will be `"off"` when this is called, right?

**Comment by twiecki** [comment]
Closing because of age. Feel free to reopen if you want to fix this up.

**Pull request closed without merging**